client/webserver/passbytes_test.go

```diff
@@ -0,0 +1,91 @@
+package webserver
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/rand"
+	"testing"
+	"time"
+)
+
+func TestPasswordUmarshallingAsBytes(t *testing.T) {
+	checkDirectUnmarshal := func(password string) error {
+		// marshal to json first
+		pwJSON, err := json.Marshal(password)
+		if err != nil {
+			return err
+		}
+		// unmarshal back from json
+		var pwFromJSON PassBytes
+		if err = json.Unmarshal(pwJSON, &pwFromJSON); err != nil {
+			return err
+		}
+		// confirm accuracy of unmarshal
+		if password != pwFromJSON.String() {
+			return fmt.Errorf("original pw: %s \tmarshalled to %s \t unmarshalled to %s", password,
+				string(pwJSON), pwFromJSON)
+		}
+		return nil
+	}
+
+	checkUnmarshalAsBodyProperty := func(password string) error {
+		// marshal to json first
+		originalData := struct {
+			Password string `json:"pass"`
+		}{
+			Password: password,
+		}
+		jsonBody, err := json.Marshal(originalData)
+		if err != nil {
+			return err
+		}
+		// unmarshal back from json
+		var restoredData = new(struct {
+			Password PassBytes `json:"pass"`
+		})
+		if err = json.Unmarshal(jsonBody, restoredData); err != nil {
+			return err
+		}
+		// confirm accuracy of unmarshal
+		if password != restoredData.Password.String() {
+			return fmt.Errorf("original body: %v \tmarshalled to %s \t unmarshalled to %v", originalData,
+				string(jsonBody), restoredData)
+		}
+		return nil
+	}
+
+	var err error
+	passwordUnmarshalledCorrectly := func(password string) bool {
+		if err = checkDirectUnmarshal(password); err != nil {
+			return false
+		}
+		err = checkUnmarshalAsBodyProperty(password)
+		return err == nil
+	}
+
+	for i := 0; i < 100; i++ {
+		if ok := passwordUnmarshalledCorrectly(randomPassword()); !ok {
```

Comment:
Randomization is good, but make sure we hit any known edges too, e.g. empty string